src/inausoft.netCLI/MappingEntry.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace inausoft.netCLI
+{
+    public class CommandInfo
+    {
+        public CommandAttribute Command { get; internal set; }
+
+        public IEnumerable<OptionAttribute> Options { get; internal set; }
+    }
+
+    internal class MappingEntry
+    {
+        public Type CommandType { get; set; }
+
+        public Type HandlerType { get; set; }
+
+        public object HandlerInstance { get; set; }
+    }
+
+    public class Mapping
+    {
+        internal List<MappingEntry> Entries { get; set; }
+
+        public Mapping()
+        {
+            Entries = new List<MappingEntry>();
+        }
+
+        public Mapping Map<T1, T2>() where T1 : class where T2 : CommandHandler<T1>
```

Comment:
I suggest using more descriptive naming for `T1`, `T2` type identifiers. Consider `TCommand` and `THandler`?

Reply:
Agreed.